src/TimeoutMigrationTool/ASQ/ASQEndpointMigrator.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace Particular.TimeoutMigrationTool.ASQ
+{
+    using System;
+    using Microsoft.Azure.Cosmos.Table;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Cosmos.Table.Queryable;
+
+    public class ASQEndpointMigrator : ITimeoutsTarget.IEndpointTargetBatchMigrator
+    {
+        public ASQEndpointMigrator(CloudTableClient client, string destinationTimeoutTableName, string stagingTableName)
```

Comment:
Most other sources and targets currently manage the connection internally. Why did you deviate from this here?

Reply:
The migrator is not the Target, the target already needs a connection. Why should it be created twice?

Reply:
Yeah right. Sorry I overlooked that